packages/disploy/src/structs/TextInputComponent.ts

```diff
@@ -0,0 +1,34 @@
+import {ModalActionRowComponent} from "./ModalActionRowComponent";
+import {ComponentType, TextInputStyle} from "discord-api-types/v10";
+
+export class TextInputComponent extends ModalActionRowComponent {
+    public style?: TextInputStyle;
+    public label?: string;
+    public minLength: number | null = null;
+    public maxLength: number | null = null;
+
+    setStyle(style: TextInputStyle){
+        this.style = style;
+        return this;
+    }
+
+    setLabel(label: string){
+        this.label = label;
+        return this;
+    }
+
+    setMinLength(length: number){
+        this.minLength = length;
+        return this;
+    }
+
+    setMaxLength(length: number){
+        this.maxLength = length;
+        return this;
+    }
+
+    constructor() {
```

Comment:
Don't force users to use the builder pattern, let them pass an object to fill the class.